### ISSUE TITLE
Broken with Ember 2.3.0-beta.2

### ISSUE THREAD

**Issue by knownasilya** [firebase/emberfire#340]
I get the following error when using with Ember 2.3.0-beta.2 (using master branch of this addon)

> TypeError: Cannot define property:container, object is not extensible.
at Function.defineProperty (native)
    at injectDeprecatedContainer (http://localhost:4200/assets/vendor.js:12085:12)
    at instantiate (http://localhost:4200/assets/vendor.js:12076:9)
    at lookup (http://localhost:4200/assets/vendor.js:11908:17)
    at Object.Container.lookup (http://localhost:4200/assets/vendor.js:11838:14)
    at fakeContainer \[as lookup\] (http://localhost:4200/assets/vendor.js:43044:56)
    at injectedPropertyGet (http://localhost:4200/assets/vendor.js:27306:18)
    at Descriptor.ComputedPropertyPrototype.get (http://localhost:4200/assets/vendor.js:25340:28)
    at Object.get (http://localhost:4200/assets/vendor.js:30658:19)
    at exports.default._emberMetalMixin.Mixin.create.get (http://localhost:4200/assets/vendor.js:45021:38)

**Comment by tstirrat** [comment]
Thanks for the report. I'm going to assume this is a bug in  2.3.0 given that it looks like some code that is meant to maintain backwards compat.

**Comment by tstirrat** [comment]
Did 2.3.0.beta2 get removed? it is not listed as a release in the ember/data repo.

**Comment by knownasilya** [comment]
`bower info ember` returns the correct thing:

> Available versions:
  - 2.3.0-beta.2
  - 2.3.0-beta.1

**Comment by tstirrat** [comment]
It now says this:

```
Available versions:
  - 2.3.0-beta.1
  - 2.3.0-beta.1
  - 2.2.1
```

I am guessing they yanked the beta.2 version

**Comment by tstirrat** [comment]
my bad, was looking at ember-data!

**Comment by rwjblue** [comment]
We are injecting `.container` into things looked up from the container.  My guess is that https://github.com/firebase/emberfire/blob/master/addon/services/firebase.js#L3-L10 is returning an instance of `Firebase`, and that it is likely `Object.freeze`'ed or `Object.seal`'ed.

**Comment by tstirrat** [comment]
Ah, thanks for this!

**Comment by rwjblue** [comment]
We can do a better job in Ember on this I think.  We should check to see if `'container' in object` before assignment.  We currently just assume all will work well, except when this happens :smiling_imp:.

**Comment by rwjblue** [comment]
Submitted https://github.com/emberjs/ember.js/pull/12699 to address.

**Comment by tstirrat** [comment]
:+1:

**Comment by knownasilya** [comment]
Seems to be working with `ember#canary`.

**Comment by rwjblue** [comment]
Just pushed the fix to beta, so `ember#beta` should be fixed soon.  It should be included in 2.3.0-beta.3 today/tomorrow.

**Comment by tstirrat** [comment]
beta.3 is out
